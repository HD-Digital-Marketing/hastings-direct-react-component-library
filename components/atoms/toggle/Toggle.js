import React, { useState } from 'react'
import { css } from 'glamor'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './toggle.scss'

const Toggle = props => {

    const {
        className = '',
        theme = '',
        nameLeft,
        nameRight,
        showState = false,
        singleChoice = false,
        id,
        onChange,
        checked = false
    } = props

    const toggleClass = classNames('a-toggle', className, {
        'a-toggle--on-dark': theme === 'dark'
    })



    let inlineCss = css({
        ':first-of-type': {
            'opacity': checked ? '0.4' : '1'
        },
        ':last-of-type': {
            'opacity': checked ? '1' : '0.4'
        },
    })

    let wrapCss = {
        backgroundColor: checked ? '#0069cc' : '#afbdd1'
    }

    return (
        <div className={toggleClass}>
            <label htmlFor={id} className="a-toggle__label">
                {nameLeft ? <span className="a-toggle__value" {...inlineCss}>{nameLeft}{showState ? (checked ? ' disabled' : ' enabled') : null}</span> : null}
                <span className="a-toggle__slider-wrap" style={singleChoice ? wrapCss : null}>
                    <input type="checkbox" id={id} name={id} defaultChecked={checked} {...{onChange}} />
                <span className="a-toggle__slider"></span>
                </span>
                {nameRight ? <span className="a-toggle__value" {...inlineCss}>{nameRight}{showState ? (!checked ? ' disabled' : ' enabled') : null}</span> : null}
            </label>
        </div>
    )
}

Toggle.propTypes = {
    className: PropTypes.string,
    theme: PropTypes.string,
    id: PropTypes.string,
    nameLeft: PropTypes.string,
    nameRight: PropTypes.string,
    checked: PropTypes.bool
}

export default Toggle
