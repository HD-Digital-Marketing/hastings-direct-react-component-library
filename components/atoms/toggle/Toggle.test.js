import React from 'react'
import renderer from 'react-test-renderer'
import Toggle from './Toggle'

describe('Check the toggle renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Toggle theme="dark" nameLeft="Annually" nameRight="Monthly" id="group" />)
            .toJSON()
        expect(tree).toMatchSnapshot()

    })
})