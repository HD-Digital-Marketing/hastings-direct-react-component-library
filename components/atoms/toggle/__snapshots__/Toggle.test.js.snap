// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the toggle renders it should match the snapshot 1`] = `
<div
  className="a-toggle a-toggle--on-dark"
>
  <label
    className="a-toggle__label"
    htmlFor="group"
  >
    <span
      className="a-toggle__value"
      data-css-1b74rh9=""
    >
      Annually
    </span>
    <span
      className="a-toggle__slider-wrap"
      style={null}
    >
      <input
        defaultChecked={false}
        id="group"
        name="group"
        type="checkbox"
      />
      <span
        className="a-toggle__slider"
      />
    </span>
    <span
      className="a-toggle__value"
      data-css-1b74rh9=""
    >
      Monthly
    </span>
  </label>
</div>
`;
